common/db/tag_store/dynamo.go

```diff
@@ -39,24 +46,33 @@ func (d *DynamoTagStore) Clear() error {
 }
 
 func (d *DynamoTagStore) Insert(tag *models.Tag) error {
+	// ensure we perform a consistent read after a write
+	defer d.setConsistentRead(true)
+
 	tag.TagID = time.Now().UnixNano()
 	return d.table.Put(tag).Run()
 }
 
 func (d *DynamoTagStore) Delete(tagID int64) error {
+	// ensure we perform a consistent read after a delete
+	defer d.setConsistentRead(true)
 	return d.table.Delete("TagID", tagID).Run()
 }
 
 func (d *DynamoTagStore) SelectAll() (models.Tags, error) {
+	defer d.setConsistentRead(false)
+
 	tags := models.Tags{}
-	if err := d.table.Scan().Consistent(true).All(&tags); err != nil {
+	if err := d.table.Scan().Consistent(d.consistentRead).All(&tags); err != nil {
 		return nil, err
 	}
 
 	return tags, nil
 }
 
 func (d *DynamoTagStore) SelectByQuery(entityType, entityID string) (models.Tags, error) {
+	defer d.setConsistentRead(false)
+
 	scan := d.table.Scan()
```

Comment:
but i think this graph pretty much indicates that this isn't going to be scalable at all:
<img width="1026" alt="screencapture at thu may 4 01 19 01 pdt 2017" src="https://cloud.githubusercontent.com/assets/97470/25695382/ba9c8cd2-3067-11e7-9e0f-d27a8be3a4b6.png">